// nbootstrap/install.js
let supportNanoapp = false;

function Copy(container) {
  let text = document.createElement('textarea')
  console.log(container.innerText)
  document.body.appendChild(text)
  text.innerText = container.innerText
  text.select()
  document.execCommand('copy')
  document.body.removeChild(text)
}

function buildScriptDOM(programs) {
  let d = document.createElement('p')
  const origin = location.origin;
  let os = getOS();
  switch (os) {
  case "linux":
    {
      supportNanoapp = true;
      let p = origin + `/nanoapps/${os}/install.sh`;
      d.innerText = `wget -O - ${p} | bash -s - ${origin}`;
      break;
    }
  case "windows":
    {
      supportNanoapp = true;
      let p = origin + programs["windows/amd64"].path
      // let p = origin + `/nanoapps/${os}/install.bat`;
      let a = document.createElement('a');
      a.appendChild(document.createTextNode("nanoapp.exe"));
      a.title = "下载地址";
      a.href = p;
      a.download="nanoapp.exe";
      d.appendChild(a)
      break;
    }
  default:
    supportNanoapp = false;
    d.innerText = `抱歉nanoapp当前不支持您的操作系统(${os})`
    break;
  }

  let button = document.createElement('button')
  button.innerText = '复制'
  button.onclick = function() {
    Copy(d)
    button.innerText = '复制成功!'
    setTimeout(() => {
      button.innerText = '复制'
    }, 3000)
  }

  let div = document.createElement('div')
  div.appendChild(d)
  div.appendChild(button)

  return div;
}

export function ForceInstall(checker, programs) {
  const lithium = whetherLithium()
  const scriptDOM = buildScriptDOM(programs)
  if (lithium) {
    LithiumInstall(checker, scriptDOM)
  } else {
    BrowserInstall(checker, scriptDOM)
  }
}

const InstallPrompt = "当前系统未安装或未激活nanoapp平台，请按照以下提示进行安装或联系管理员协助安装。"

let nanoapp_iframe_failed = false
function rewritePageToNanoappInstaller(prompt, scriptDOM) {
  if (nanoapp_iframe_failed) { //无法访问 外网的nanoapp.linakes.com
    document.body.innerText = " 请联系管理员下载并安装nanoapp.bootstrap"
    return
  }

  const guard = "__nanoapp_installer_guard";
  if (document.getElementById(guard)) {
    return
  }
  document.body.innerHTML = '';
  let title = document.createElement('div');
  title.innerText = prompt;
  title.id = guard;
  document.body.appendChild(title);

  document.body.appendChild(scriptDOM)
}

function insertOfficalWebSite() {
  const url = "http://nanoapp.linakesi.com/";
  let ifrm = document.createElement("iframe");
  ifrm.setAttribute("src", url);
  ifrm.style.width = "100%";
  ifrm.style.height = "100%";

  // TODO:找到如何检测网络错误的方式
  ifrm.onerror = function() {
    alert("加载失败")
    nanoapp_iframe_failed = true;
  }
  document.body.appendChild(ifrm);
}

function getOS() {
  let userAgent = navigator.userAgent,
      platform = navigator.platform,
      macosPlatforms = ['Macintosh', 'MacIntel', 'MacPPC', 'Mac68K'],
      windowsPlatforms = ['Win32', 'Win64', 'Windows', 'WinCE'],
      iosPlatforms = ['iPhone', 'iPad', 'iPod'],
      os = null;

  if (macosPlatforms.indexOf(platform) !== -1) {
    os = 'Mac OS';
  } else if (iosPlatforms.indexOf(platform) !== -1) {
    os = 'iOS';
  } else if (/Android/.test(userAgent)) {
    os = 'Android';
  } else if (windowsPlatforms.indexOf(platform) !== -1) {
    os = 'windows';
  } else if (!os && /Linux/.test(platform)) {
    os = 'linux';
  }
  return os;
}

function LithiumInstall(checker, scriptDOM) {
  rewritePageToNanoappInstaller(InstallPrompt, scriptDOM)
  const reload = location.reload.bind(location);

  if (supportNanoapp && confirm("是否自动为本机安装nanoapp平台?")) {
    try {
      lnks.$(scriptDOM.innerText, LithiumInstallErrorDom, console.log)
    } catch (e) {
      console.error("catch error:", e)
      LithiumInstallErrorDom(e)
    }
  }

  const id = setInterval(async function() {
    rewritePageToNanoappInstaller(InstallPrompt, scriptDOM)//避免被其他js覆盖
    if (await checker()) {
      clearInterval(id);
      setTimeout(reload, 1000) //延迟1s刷新
    }
  }, 1000)
}

function LithiumInstallErrorDom (err) {
  let p = document.createElement("p")
  p.innerText = `自动安装Bootstrap失败，Linux环境下，请手动复制上面的命令到终端执行，Windows环境，请手动下载对应的执行程序并执行安装。如果还是不能正常工作，可以联系管理员进行协助安装。相关错误信息如下: ${err}`

  document.body.appendChild(p)
}

function BrowserInstall(checker, scriptDOM) {
  rewritePageToNanoappInstaller(InstallPrompt, scriptDOM)//避免被其他js覆盖
  // TODO: 如果可以*访问互联网*，则直接将http://nanoapp.linakesi.com/install作为iframe，内嵌到当前页面。

  const reload = location.reload.bind(location);

  const id = setInterval(async function() {
    rewritePageToNanoappInstaller(InstallPrompt, scriptDOM)//避免被其他js覆盖
    if (await checker()) {
      clearInterval(id);
      setTimeout(reload, 1000) //延迟1s刷新
    }
  }, 1000)
}

function whetherLithium() {
  try {
    return lnks.$
  } catch(e) {
    return false;
  }
}
